source/Handlebars/Compiler/Lexer/Converter/BlockAccumulators/BlockAccumulatorContext.cs

```diff
@@ -29,6 +29,10 @@ public static BlockAccumulatorContext Create(Expression item, ICompiledHandlebar
             {
                 context = new BlockHelperAccumulatorContext(item);
             }
+            else if (IsLooseClosingElement(item, parentItem, out var looseBlockName))
+            {
+                throw new HandlebarsCompilerException($"Loose closing block '{looseBlockName}' was found");
```

Comment:
Dropping this in from my mobile to start a conversation; I can follow up with more specific suggestions: I am not sure the exception message or phrase “loose closing block” is clear or actionable enough by itself for many devs. They would need to look that up to understand it. Can we make it more clear what the problem is?

Reply:
I see your point and I had worries about the wording. What do you think about "An open element was missed for the closing element '/unless'"?

Reply:
Now that we're adding more specific compile-time messages, I'm really wishing I'd added the ability to keep line number and source context :) 

"A closing element {{/unless}} was found without a matching open element"?

Reply:
> Now that we're adding more specific compile-time messages, I'm really wishing I'd added the ability to keep line number and source context :)

@rexm , I've added this some time ago but unfortunately it's not propagated to every corner of the compiler 🙁

Reply:
Oh awesome! I’ll need to look at how that works :)

Reply:
The message was updated. Also, I've made renaming for functions and vars to get rid of some ambiguity.